providers/domain.rb

```diff
@@ -38,17 +38,22 @@
       cmd << " -DomainName #{new_resource.name}"
       cmd << " -SafeModeAdministratorPassword (convertto-securestring '#{new_resource.safe_mode_pass}' -asplaintext -Force)"
       cmd << ' -Force:$true'
+      cmd << ' -NoRebootOnCompletion' if !new_resource.restart
     else node[:os_version] <= '6.1'
       cmd = 'dcpromo -unattend'
       cmd << " -newDomain:#{new_resource.type}"
       cmd << " -NewDomainDNSName:#{new_resource.name}"
-      cmd << ' -RebootOnCompletion:Yes'
+      if !new_resource.restart
```

Comment:
Any reason to make this a Not and reverse lines 47 and 49?

Seems to read cleaner than how it currently is written, but see no operational issue either way.